pkg/registry/backgroundsvcs/adapter/service_test.go

```diff
@@ -0,0 +1,213 @@
+package adapter
+
+import (
+	"context"
+	"errors"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/grafana/dskit/services"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	// Constants for timeout-based tests
+	testContextTimeout  = 50 * time.Millisecond
+	expectedMinDuration = 45 * time.Millisecond
+)
+
+// Mock service for testing
+type mockService struct {
+	mock.Mock
```

Comment:
:pray: Can we do without `mock.Mock`? It makes tests more difficult to read and adapt. (Feel free to ignore, I just don't like seeing it in new code.)

Reply:
My main issue with `mock.Mock` is use of strings to setup expectations/calls, where simple Go code can do as well. Your replacement looks great to me -- it's simple to understand and follow. Thank you!